x/evm/types/params.go

```diff
@@ -186,7 +197,74 @@ func validateEIP712AllowedMsgs(i interface{}) error {
 	return nil
 }
 
+func validateEnabledPrecompiles(enabledPrecompiles []string) error {
+	for _, addr := range enabledPrecompiles {
+		if !common.IsHexAddress(addr) {
+			return fmt.Errorf("invalid hex address: %v in enabled precompiles list", addr)
+		}
+	}
+
+	addrs := make([]common.Address, len(enabledPrecompiles))
+	for i, precompile := range enabledPrecompiles {
+		addrs[i] = common.HexToAddress(precompile)
+	}
+
+	if err := checkIfSortedInBytesRepr(addrs); err != nil {
+		return fmt.Errorf("enabled precompiles are not sorted: %v", err)
+	}
+
+	if err := checkIfUniqueInBytesRepr(addrs); err != nil {
+		return fmt.Errorf("enabled precompiles are not unique: %v", err)
+	}
+
+	return nil
+}
+
+func checkIfSortedInBytesRepr(addrs []common.Address) error {
+	n := len(addrs)
+
+	for i := 0; i < n-1; i++ {
+		cmp := bytes.Compare(addrs[i].Bytes(), addrs[i+1].Bytes())
+		if cmp == 1 {
+			return fmt.Errorf("addresses are not sorted, %v > %v", addrs[i].Hex(), addrs[i+1].Hex())
+		}
+	}
+
+	return nil
+}
+
+func checkIfUniqueInBytesRepr(addrs []common.Address) error {
+	n := len(addrs)
+
+	exists := make(map[common.Address]struct{}, n)
+	for _, addr := range addrs {
+		if _, ok := exists[addr]; ok {
+			return fmt.Errorf("addr %v not unique", addr.Hex())
+		}
+
+		exists[addr] = struct{}{}
+	}
+
+	return nil
+}
+
 // IsLondon returns if london hardfork is enabled.
 func IsLondon(ethConfig *params.ChainConfig, height int64) bool {
 	return ethConfig.IsLondon(big.NewInt(height))
 }
+
+func CheckIfEnabledPrecompilesAreRegistered(registeredModules []precompile_modules.Module, enabledPrecompiles []string) error {
```

Comment:
need comment on public method

also maybe a name like `ValidateEnabledPrecompiles` or `ValidatePrecompileRegistration` is more idiomatic (and future proof?)

Reply:
agree, `ValidatePrecompileRegistration` sounds better

Reply:
Fixed: https://github.com/Kava-Labs/ethermint/pull/55/commits/c2e188f64f76a433679d10b59e47131ef62ce35d